### ISSUE TITLE
VestaCP stop resolving names on fresh install (LXC Container)

### ISSUE THREAD

**Issue by amvidalrc** [serghey-rodin/vesta#1518]
### Operating System (OS/VERSION):

Debian 9, CentOS 7

### VestaCP Version:

vesta/stable,now 0.9.8-19 amd64 [installed]

### Installed Software (what you got with the installer):

Full package (Including PostgreSQL, Remi) without Quota

### Steps to Reproduce:

Over Proxmox:
 - New LXC Container with Debian 9
 - Ping Ok
 - apt update
 - apt upgrade -y
 - apt install curl -y
 - curl -O http://vestacp.com/pub/vst-install.sh
 - bash vst-install.sh --nginx yes --apache yes --phpfpm no --named yes --remi yes --vsftpd yes --proftpd no --iptables yes --fail2ban yes --quota yes --exim yes --dovecot yes --spamassassin yes --clamav yes --softaculous yes --mysql yes --postgresql yes

 - When installation ends, ping any host, not working. resolv.conf its OK. Ping to any IP its ok:

root@vestaCP:~# ping google.com
^C
root@vestaCP:~# ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
64 bytes from 8.8.8.8: icmp_seq=1 ttl=59 time=5.00 ms
64 bytes from 8.8.8.8: icmp_seq=2 ttl=59 time=5.02 ms
^C
--- 8.8.8.8 ping statistics ---
2 packets transmitted, 2 received, 0% packet loss, time 1001ms
rtt min/avg/max/mdev = 5.008/5.016/5.024/0.008 ms

### Related Issues/Forum Threads:

NO found nothing.

### Other Notes:

Nothing 👍

**Comment by amvidalrc** [comment]
This is the last step that Vesta Installer do correctly:

**Comment by Skamasle** [comment]
In that step vesta configure firewall and fail2ban

Check your firewall rules, check your LXC and check if your node have all iptables modules

**Comment by amvidalrc** [comment]
I have checked and everything its OK. I have old proxmox running OK, its happening only with the latest vestacp version.

**Comment by amvidalrc** [comment]
If I install without iptables and fail2ban, everything goes fine.

**Comment by jeret15** [comment]
I had the same problem on my Ubuntu 16.04 container.
Upgrading the Proxmox pve kernel (to version 4.15.17-8) on the hypervisor seems to have fixed the problem for me.

**Comment by luizjrdeveloper** [comment]
I have the same problem.
LXC with Ubuntu 16.04 I installed the Latest Version.
But after that to ping ping or external connections

**Comment by luizjrdeveloper** [comment]
@amvidalrc Hello.
Today I have already solved this problem here.
I am using Proxmox in OVH, I created an LXC and before starting everything I added permission in Firewall Container (LXC).

Notice how it was in the picture.
![captura de tela 2018-05-09 as 16 06 28](https://user-images.githubusercontent.com/1474563/39836507-07165dfc-53aa-11e8-9f36-9b39d3070f5c.png)

**Comment by trestylez** [comment]
I just experienced the same issue. Looks like the iptable rules does not allow external DNS lookups, so if you use an external DNS you should get the IP from /etc/resolve.conf and add the following rules which  fixed the issue for me:
iptables -A OUTPUT -p udp -d <IP> --dport 53 -m state --state NEW,ESTABLISHED -j ACCEPT
iptables -A INPUT  -p udp -s <IP> --sport 53 -m state --state ESTABLISHED     -j ACCEPT
iptables -A OUTPUT -p tcp -d <IP> --dport 53 -m state --state NEW,ESTABLISHED -j ACCEPT
iptables -A INPUT  -p tcp -s <IP> --sport 53 -m state --state ESTABLISHED     -j ACCEPT

**Comment by jodumont** [comment]
I **had** the same issue with

VestaCP 0.9.8 on a Debian 9 turning as a unprivileged container under Proxmox 5.2.1

### VestaCP is installed with this command :

> bash vst-install.sh --nginx yes --phpfpm yes --apache no   --mysql yes \\
> --iptables yes --fail2ban yes --quota yes \\
> --exim no --dovecot no --spamassassin no --clamav no \\
> --softaculous no --named no --remi no \\
> --vsftpd no --proftpd no \\
> --postgresql no \\
> --hostname \\
>  --email \\
>  --password 

### After finishing the install, before a reboot;
1. PING is able to resolve `www.com` and `ttt.com` (yes they are real website)
2. after the reboot PING is unable to resolve `www.com` and `ttt.com` but still working for IP like `8.8.8.8`

### How I resolve it ?
> Have you tried turning it off and on again ? will said Roy

So after reinstalling again and over, searching and reading, sleeping and drinking coffee, ...
I just when to the VestaCP Interface and turn the IPtables off and on'

> IPtables off == yes it PING
> IPtables on again == yes it PING `www.com` and `ttt.com`

And magie; it still resolve even after a reboot.
and it's not just a cache DNS history because it's resolve `eee.com` too :+1: 

-- So Long, and Thanks for All the Fish.

**Comment by jodumont** [comment]
@trestylez where did you execute these ?

1. on the vesta guest or directly on the proxmox host ?

2. your container was privileged ?

**Comment by jodumont** [comment]
still investigating on this, 
+ @luizjrdeveloper; I tried your trick without success, but maybe it's because my container is unprivileged

anyway I found this message which probably point in the right direction

just after installing **fail2ban**
> --2018-07-27 13:31:10--  http://c.vestacp.com/debian/9/fail2ban.tar.gz
> Resolving c.vestacp.com (c.vestacp.com)... 104.236.66.100
> Connecting to c.vestacp.com (c.vestacp.com)|104.236.66.100|:80... connected.
> HTTP request sent, awaiting response... 200 OK
> Length: 712 [application/octet-stream]
> Saving to: 'fail2ban.tar.gz'
> fail2ban.tar.gz     100%[=================>]     712  --.-KB/s    in 0s      
> 2018-07-27 13:31:10 (177 MB/s) - 'fail2ban.tar.gz' saved [712/712]
> 

errors start with :

> chattr: Permission denied while setting flags on /home/admin/conf
> mv: cannot move '/usr/local/vesta/data/ips/' to a subdirectory of itself, '/usr/local/vesta/data/ips/192.168.1.8'
> sed: -e expression #1, char 0: no previous regular expression
> mount: cannot remount rpool/data/subvol-8888-disk-1 read-write, is write-protected

than after this vesta start to being unable to resolve including himself
> sudo: unable to resolve host vesta.dev.localdomain: Connection timed out

**Comment by dpeca** [comment]
No, it's another 'bug' - https://github.com/serghey-rodin/vesta/commit/f68bb795deb153bbb0b9da3a03b0f7377455bee9#commitcomment-29626998
I believe it's not related with this issue, that is described in this topic.

**Comment by trestylez** [comment]
@jodumont unprivileged container and DNS iptable rules run on the container itself.

**Comment by jodumont** [comment]
#### just to be sure;
vesta works perfectly as a unprivileged lxc container debian when iptables is disabled

> bash vst-install.sh --nginx yes --phpfpm yes --apache no --named yes --remi no --vsftpd yes --proftpd no --iptables yes --fail2ban yes --quota no --exim no --dovecot no --spamassassin no --clamav no --softaculous no --mysql yes --postgresql no

than after the install `update-rc.d -f iptables remove`

#### and _FYI_ no i'm not that c**r**_az_**y**;

> I have a frontend firewall (pfSense)
> who allow only http/https on this container.

**Comment by marph86** [comment]
I have unprivileged LXC. Malfunctioning FW is wrong..
ISPconfig with iptables + Fail2ban works correctly.
I wanted to exchange ISPconfig for VESTACP.
Thank you for repairing:)

**Comment by Azzazello1979** [comment]
> I just experienced the same issue. Looks like the iptable rules does not allow external DNS lookups, so if you use an external DNS you should get the IP from /etc/resolve.conf and add the following rules which fixed the issue for me:
> iptables -A OUTPUT -p udp -d --dport 53 -m state --state NEW,ESTABLISHED -j ACCEPT
> iptables -A INPUT -p udp -s --sport 53 -m state --state ESTABLISHED -j ACCEPT
> iptables -A OUTPUT -p tcp -d --dport 53 -m state --state NEW,ESTABLISHED -j ACCEPT
> iptables -A INPUT -p tcp -s --sport 53 -m state --state ESTABLISHED -j ACCEPT

trestyley can you please elaborate on what you are doing here. I have external dns so I unchecked the the "manage DNS" when I added domains. My problem is what you describe here, Looks like the iptable rules does not allow external DNS lookups. However the commands gave me errors in sh: Bad argument `53'. Also what do you mean by you should get the IP from /etc/resolve.conf... ?
Do I have to append that file? Get the ip from that file and do what with them? There are two nameserver ip addresses in that file. Thank you!

**Comment by devrimerduman** [comment]
Hi, I want to install vestacp lxc on proxmox as well. can someone share installation process please, thanks.

**Comment by Neustradamus** [comment]
Dear all,

@Dennis-SEG has done a very good job here:
- https://github.com/outroll/vesta/pull/2322

What do you think?
